Mathlib/Algebra/Group/Submonoid/Defs.lean

```diff
@@ -458,23 +458,6 @@ instance toMonoid {M : Type*} [Monoid M] (S : Submonoid M) : Monoid S := fast_in
 instance toCommMonoid {M} [CommMonoid M] (S : Submonoid M) : CommMonoid S := fast_instance%
   Subtype.coe_injective.commMonoid Subtype.val rfl (fun _ _ => rfl) fun _ _ => rfl
 
-/-- A submonoid of a left cancellative unital magma inherits left cancellation. -/
-@[to_additive
-      "An `AddSubmonoid` of a left cancellative unital additive magma inherits left cancellation."]
-instance isLeftCancelMul [IsLeftCancelMul M] (S : Submonoid M) : IsLeftCancelMul S :=
-  Subtype.coe_injective.isLeftCancelMul Subtype.val fun _ _ => rfl
-
-/-- A submonoid of a right cancellative unital magma inherits right cancellation. -/
-@[to_additive
-      "An `AddSubmonoid` of a right cancellative unital additive magma inherits right
-      cancellation."]
-instance isRightCancelMul [IsRightCancelMul M] (S : Submonoid M) : IsRightCancelMul S :=
-  Subtype.coe_injective.isRightCancelMul Subtype.val fun _ _ => rfl
-
-/-- A submonoid of a cancellative unital magma inherits cancellation. -/
-@[to_additive "An `AddSubmonoid` of a cancellative unital additive magma inherits cancellation."]
-instance isCancelMul [IsCancelMul M] (S : Submonoid M) : IsCancelMul S where
-
```

Comment:
@Vierkantor or @j-loreaux, do you remember if we discovered it was better to duplicate these instances for MulMemClass and the concrete instantiations? Certainly we still seem to do so for the `Monoid` instance above.